ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/TableExpireService.java

```diff
@@ -168,6 +173,17 @@ public void run() {
     }
   }
 
+  public static long getExpireTime(UnkeyedTable table, long olderThan) {
+    ArrayList<Snapshot> snapshots = Lists.newArrayList(table.snapshots());
```

Comment:
How about using Iterator to iterate the snasphots? it could reduce the time complex to `n`, but now it is `n to 2n`

Reply:
It makes sense. I will modify it.